pkg/skaffold/test/custom/custom.go

```diff
@@ -0,0 +1,144 @@
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+	"runtime"
+	"time"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/list"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/color"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+// for tests
+var doRunCustomCommand = runCustomCommand
+
+const Windows string = "windows"
+
+type Runner struct {
+	customTest     latest.CustomTest
+	testWorkingDir string
+}
+
+// New creates a new custom.Runner.
+func New(cfg docker.Config, wd string, ct latest.CustomTest) (*Runner, error) {
+	return &Runner{
+		customTest:     ct,
+		testWorkingDir: wd,
+	}, nil
+}
+
+// Test is the entrypoint for running custom tests
+func (ct *Runner) Test(ctx context.Context, out io.Writer, _ []build.Artifact) error {
+	if err := doRunCustomCommand(ctx, out, ct.customTest); err != nil {
+		return fmt.Errorf("running custom test command: %w", err)
+	}
+
+	return nil
+}
+
+func runCustomCommand(ctx context.Context, out io.Writer, test latest.CustomTest) error {
+	// Expand command
+	command, err := util.ExpandEnvTemplate(test.Command, nil)
+	if err != nil {
+		return fmt.Errorf("unable to parse test command %q: %w", test.Command, err)
+	}
+
+	if test.TimeoutSeconds <= 0 {
+		color.Default.Fprintf(out, "Running custom test command: %q\n", command)
+	} else {
+		color.Default.Fprintf(out, "Running custom test command: %q with timeout %d s\n", command, test.TimeoutSeconds)
+		newCtx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+		defer cancel()
+		ctx = newCtx
+	}
+
+	var cmd *exec.Cmd
+	// We evaluate the command with a shell so that it can contain env variables.
+	if runtime.GOOS == Windows {
+		cmd = exec.CommandContext(ctx, "cmd.exe", "/C", command)
+	} else {
+		cmd = exec.CommandContext(ctx, "sh", "-c", command)
+	}
+
+	cmd.Stdout = out
+	cmd.Stderr = out
+
+	if err := util.RunCmd(cmd); err != nil {
+		if e, ok := err.(*exec.ExitError); ok {
+			// If the process exited by itself, just return the error
+			if e.Exited() {
+				color.Red.Fprintf(out, "Command finished with non-0 exit code.\n")
+				return fmt.Errorf("command finished with non-0 exit code: %w", e)
+			}
+			// If the context is done, it has been killed by the exec.Command
+			select {
+			case <-ctx.Done():
+				if ctx.Err() == context.DeadlineExceeded {
+					color.Red.Fprintf(out, "Command timed out\n")
+				} else if ctx.Err() == context.Canceled {
+					color.Red.Fprintf(out, "Command cancelled\n")
+				}
+				return ctx.Err()
+			default:
+				return e
+			}
+		}
+		return err
+	}
+	color.Green.Fprintf(out, "Command finished successfully\n")
+
+	return nil
+}
+
+// TestDependencies returns dependencies listed for a custom test
+func (ct *Runner) TestDependencies() ([]string, error) {
```

Comment:
This should receive a `context.Context`, just like in `Test()` above.  Then you can replace `context.Background()` below.

Reply:
Custom method implements this interface
https://github.com/GoogleContainerTools/skaffold/blob/31cf2f13ae2419c558ec12c6ccf60567f388e2a2/pkg/skaffold/test/types.go#L33
And had to follow the method signature ` TestDependencies() ([]string, error)`.

Is there a different way to send it 'context.Context`?